src/parsing/Pfff_or_tree_sitter.ml

```diff
@@ -36,10 +36,12 @@ type 'ast parser =
   | Pfff of (Common.filename -> 'ast * Parsing_stat.t)
   | TreeSitter of (Common.filename -> 'ast Tree_sitter_run.Parsing_result.t)
 
+(* This type is parametrized by the AST type because ??? *)
```

Comment:
because we don't always generate directly generic ASTs; we sometimes generate intermediate ASTs hence the 
need for polymorphic type (so you can have Ast_cpp.program internal_result, or Ast_php.program internal_result.